serve/mlc_serve/engine/staging_engine.py

```diff
@@ -61,6 +58,11 @@ def __init__(
         self.requests_lock = Lock()
         self.requests = dict[RequestId, RequestState]()
 
+        # TODO(@team): This is a temporary solution to expose model config to higher API layer.
+        #   Follow-up with the proper solution
+        self.model_artifact_config = get_model_artifact_config(
```

Comment:
We should make the `ModelConfig` a property of all engines and pass it in.

Reply:
It seems tricky atm since `StagingInferenceEngine` does not inherit from `EngineBase`. @yelite, can we expose this during your refactoring?